assets/src/apps/app.tsx

```diff
@@ -1,12 +1,47 @@
 import { ModalDisplay } from 'components/modal/ModalDisplay';
-import React from 'react';
+import React, { FC } from 'react';
 import ReactDOM from 'react-dom';
 import { Provider } from 'react-redux';
 import { State } from 'state/index';
 import { configureStore } from 'state/store';
 import { Maybe } from 'tsmonad';
 import { b64DecodeUnicode } from 'utils/decode';
 
+/**
+ * Registers a client side react/js application to be used in Phoenix templates via `ReactPhoenix.ClientSide.react_component`
+ * This will wrap the component in a redux store and give it a ModalDisplay top level sibling component it then registers
+ * that as window.Component.*name*
+ *
+ * @param name Name of the app to register. Is used in both the key on window.Components and the debugging label in react-dev-tools
+ * @param Component Any react component that represents a top-level applicacation
+ */
+export function registerApplication(name: string, Component: React.FunctionComponent<any>) {
```

Comment:
Be nice to resolve this todo question before wrapping up.

Reply:
I'm not sure if I can answer this one for you. I think we'll have to consult with Darren about it. I can tell you that the adaptive lesson app doesn't use this store, so it's one case already.

Reply:
If we know for sure one app doesn't need it, then that's a decent argument to move it up a level to just the apps that do.
Does adaptive lesson need the <ModalDisplay /> bit?

Reply:
I think the idea was that it should (eventually) but I don't believe it currently does.